tox.ini

```
@@ -1,57 +1,88 @@
 [tox]
 envlist =
-    py27,py35,py36,py37,py38,pypy
-    docs,{py2,py3}-cover,coverage,
+    lint,
+    py27,pypy,
+    py35,py36,py37,py38,pypy3,
+    docs,
+    coverage
+isolated_build = True
 
 [testenv]
-# Most of these are defaults but if you specify any you can't fall back
-# to defaults for others.
-basepython =
-    py27: python2.7
-    py35: python3.5
-    py36: python3.6
-    py37: python3.7
-    py38: python3.8
-    pypy: pypy
-    pypy3: pypy3
-    py2: python2.7
-    py3: python3.5
+commands =
+    python --version
+    pytest {posargs:}
+extras =
+    testing
+setenv =
+    COVERAGE_FILE=.coverage.{envname}
+
+[testenv:coverage]
+skip_install = True
+commands =
+    coverage combine
+    coverage xml
+    coverage report --fail-under=100
+deps =
+    coverage
+setenv =
+    COVERAGE_FILE=.coverage
+depends = py27, py38
 
+[testenv:lint]
+skip_install = True
 commands =
-    pip install pyramid_nacl_session[testing]
-    nosetests --with-xunit --xunit-file=nosetests-{envname}.xml {posargs:}
+    black --check --diff .
+    check-manifest
+    # flake8 src/waitress/ tests
+    # build sdist/wheel
+    python -m pep517.build .
+    twine check dist/*
+deps =
+    black
+    readme_renderer
+    check-manifest
+    pep517
+    twine
+    flake8
+    flake8-bugbear
 
 [testenv:docs]
-basepython = python3.5
-whitelist_externals = make
+whitelist_externals =
+    make
 commands =
-    pip install pyramid_nacl_session[docs]
-    make -C docs html epub BUILDDIR={envdir} "SPHINXOPTS=-W -E"
+    make -C docs html BUILDDIR={envdir} SPHINXOPTS="-W -E"
```

If this fails on Sphinx 3.0.x, then:

```suggestion
    make -C docs html BUILDDIR={envdir} SPHINXOPTS="-W -E -D suppress_warnings=ref.term"
```

---

Looks like it is fine without it, due to no conflicts in case-sensitivity in the Glossary.